manager/src/main/java/io/gingersnapproject/search/QueryHandler.java

```diff
@@ -0,0 +1,25 @@
+package io.gingersnapproject.search;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import io.gingersnapproject.Caches;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+
+@Singleton
+public class QueryHandler {
+   @Inject
+   Caches caches;
+
+   @Inject
+   SearchBackend searchBackend;
+
+   public Multi<String> query(String query) {
+      Uni<SearchResult> result = searchBackend.query(query);
+      return result.toMulti().onItem().transformToIterable(SearchResult::hits)
+            .onItem().transformToUni(sh -> caches.get(sh.indexName(), sh.documentId()))
```

Comment:
Next step improvement for this is we should add a `getIfPresent` method on `Caches` and for each entry that isn't present we need to do a batch query or IN clause from the DB. Currently this could cause up to N database selects if none of the entries are in the cache.